source/common/http/http2/codec_impl.cc

```diff
@@ -641,12 +645,17 @@ Http::Status ConnectionImpl::dispatch(Buffer::Instance& data) {
 }
 
 Http::Status ConnectionImpl::innerDispatch(Buffer::Instance& data) {
+  ScopeTrackerScopeState scope(this, connection_.dispatcher());
   ENVOY_CONN_LOG(trace, "dispatching {} bytes", connection_, data.length());
   // Make sure that dispatching_ is set to false after dispatching, even when
   // ConnectionImpl::dispatch returns early or throws an exception (consider removing if there is a
   // single return after exception removal (#10878)).
-  Cleanup cleanup([this]() { dispatching_ = false; });
+  Cleanup cleanup([this]() {
+    dispatching_ = false;
+    current_slice_ = nullptr;
+  });
```

Comment:
If we did that we'd end up ctor and dtoring the `cleanup` object every iteration of this loop instead of every call to this function. Thoughts?

Reply:
I guess construction/destruction of the callback and the indirection through the callback is far from free.  I guess this could be something to hope for in a future C++ standard.

It would be good to have a ScopedSetter class that we could use to set the members to the new value and restore the old value when exiting the scope.  Something like:

template<class T> class ScopedSetter {
public:
     ScopedSetter(T& item, T new_value) : item_(item), old_value_(item) {
       item_ = new_value
    }
    ~ScopedSetter() { item_ = old_value_; }
private:
  T& item_;
  T old_value_;
}